src/Core/GrpcTrait.php

```diff
@@ -252,7 +252,7 @@ private function formatTimestampForApi($value)
         preg_match('/\.(\d{1,9})Z/', $value, $matches);
         $value = preg_replace('/\.(\d{1,9})Z/', '.000000Z', $value);
         $dt = \DateTimeImmutable::createFromFormat('Y-m-d\TH:i:s.u\Z', $value);
-        $nanos = (isset($matches[1])) ? $matches[1] : 0;
+        $nanos = (isset($matches[1])) ? str_pad($matches[1], 9, '0') : 0;
```

Comment:
It looks like we already have a test for this function here: https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/tests/unit/Core/GrpcTraitTest.php#L216

Can we modify the test so that it catches this error? Or, do we need to add a new test?

Reply:
Great call. I'll add a data provider

Reply:
Updated. PTAL